### PR TITLE
Prepare for Issue #1040: add a fancy GitHub Action for 'shellcheck' for all pushes and PRs

### PR DESCRIPTION
**Pull request by thomasmerz**
##### Pull Request Details:

* Type: **Enhancement**

* Impact: **High**

* Reference to related issue (URL):
This is a preparation for future PRs from me to solve #1040

* How was this pull request tested?
See GitHub Action in my forked repo for this PR:
https://github.com/thomasmerz/rear/runs/5069429307?check_suite_focus=true

* Brief description of the changes in this pull request:
This PR introduces a GitHub Action that will `shellcheck` all scripts in dirs "test" and "usr" for committer's and also repo-owner's convenience. All warnings including and above "severity=warning" will be reported and made transparent.

### CODE REVIEW

**Comment by pcahyna** [comment]
@antonvoznia has been working on something similar recently, but if you beat him to it, why not :-)

**Comment by pcahyna** [comment]
That said, it looks quite primitive. Do you have an example of a run for a pull request in your repo (pull request from your branch to your branch)? I am afraid that any trivial pull request will produce hundreds if not thousands of warnings unrelated to the PR itself. I believe that what we need is to display only errors that are introduced by the PR, not everything.

**Comment by thomasmerz** [comment]
> That said, it looks quite primitive. Do you have an example of a run for a pull request in your repo (pull request from your branch to your branch)? I am afraid that any trivial pull request will produce hundreds if not thousands of warnings unrelated to the PR itself. I believe that what we need is to display only errors that are introduced by the PR, not everything.

1. You could **merge this PR not before a fix+foxy `shellcheck`ed version exists**, so that any following push/merge  to master and every PR will only `shellcheck` (all) **scripts in a specific path** (tests/** and usr/**). So we won't get any warning that won't be related to the PR itself.
Details: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-paths

2. Or/and we can also **trigger no action on pushing to HEAD or merging into HEAD** branch.

See my next [commit](https://github.com/thomasmerz/rear/commit/9c93748dd36d7dc130a757dcaa1ecd8a4bc61121) which didn't trigger an action for point 1.
 
So, do you agree to this (1) or do you think point 2 might  also be reasonable?

**Comment by jsmeix** [comment]
I won't find noticeable time for this pull request until ReaR 2.7 was released, cf.
https://github.com/rear/rear/issues/2751
so I set this pull request's milestone to ReaR v2.8 i.e. after ReaR 2.7 was released.

**Comment by thomasmerz** [comment]
Ping… just to keep this PR "active"…

**Comment by thomasmerz** [comment]
What about this PR? @jsmeix and @schlomo

**Comment by thomasmerz** [comment]
@gdha ping.

**Comment by thomasmerz** [comment]
Ping @gdha

**Review by gdha — APPROVED** [review]

File: .github/workflows/shellcheck.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+jobs:
+  build:
+    name: Shellcheck Lint
+
+    # This job runs on Linux
+    runs-on: ubuntu-latest
+
+    steps:
+      # Required to access files of this repository
+      - uses: actions/checkout@master
```
Comment (jsmeix) [inline .github/workflows/shellcheck.yml:30]: See https://github.com/rear/rear/pull/2756#discussion_r1048156831

**Review by jsmeix — CHANGES_REQUESTED** [review]

File: .github/workflows/shellcheck.yml (hunk 1 above)
Comment (thomasmerz) [reply .github/workflows/shellcheck.yml]: I agree and changed/pinned it to `v3`.

**Pull request closed without merging**